### ISSUE TITLE
Incorrect no-speech probability in `WhisperNoSpeechDetection` when `language` is set and `num_beams > 1`

### ISSUE THREAD

**Issue by cifkao** [huggingface/transformers#29313]
### System Info

- `transformers` version: 4.39.0.dev0 (`c8d9840`)

### Who can help?

@patrickvonplaten @sanchit-gandhi

### Reproduction

 In [`WhisperNoSpeechDetection`](https://github.com/huggingface/transformers/blob/c8d98405a8f7b0e5d07391b671dcc61bb9d7bad5/src/transformers/generation/logits_process.py#L1893), in some cases, the model is run again inside the logits processor to get the **logits**, but then, if `self.is_scores_logprobs == True`, those logits are run through `exp()` instead of `softmax()`, resulting in incorrect (unnormalized) probabilities (possibly `> 1`).

This scenario arises when the model is called with `language` set and with `num_beams > 1`.

### Expected behavior

The no-speech probability should be the same regardless of the value of `num_beams`.

**Comment by cifkao** [comment]
@ylacombe

**Comment by ylacombe** [comment]
Thanks for the ping @cifkao, [I've answered](https://github.com/huggingface/transformers/pull/29248#issuecomment-2029358885) directly on the PR !
